### PR TITLE
Refactor factories description and example

### LINKED ISSUE #2: non-working "reccomended" example
[y022](https://github.com/rwwagner90/angular-styleguide-es6#style-y022)

```
/* recommended */
angular
    .module('app')
    .controller('SomeController', SomeController);

class SomeController {
  constructor() { }
}
```

ES6 classes are not hoisted as I understand it, and as such, this example will not work. SomeController will not be defined here before it is used.

### PR DESCRIPTION
**Pull request by mikeerickson**
Refactored description (added notes) and code example to better reflect how factories are used with Angular 1.x and ES6.

This closes https://github.com/rwwagner90/angular-styleguide-es6/issues/2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -454,15 +454,19 @@ controllerAs can also be used in the router like so:
 
     *Why?*: Data service implementation may have very specific code to handle the data repository. This may include headers, how to talk to the data, or other services such as `$http`. Separating the logic into a data service encapsulates this logic in a single place hiding the implementation from the outside consumers (perhaps a controller), also making it easier to change the implementation.
 
+    Note: The following ES6 factory definition uses the `new` operator to instantiate the factory function, and supplying the desired services used internally.
```
Comment (RobbieTheWagner) [inline README.md]: Should this be "and is supplying"? Something is off with the grammar here

Comment (RobbieTheWagner) [inline README.md]: @mikeerickson or should this be and injects the desired services?

**Comment by RobbieTheWagner** [comment]
@mikeerickson if we can figure out the correct wording for this, I am ready to merge.

**Comment by mikeerickson** [comment]
@rwwagner90 sorry for delay, I will get this cleared up today and update PR

**Comment by mikeerickson** [comment]
@rwwagner90 Done

**Comment by RobbieTheWagner** [comment]
Merged 🎉

**Pull request merged**